runtime/tests/checker/arrays_dictionaries_test.go

```diff
@@ -753,6 +753,72 @@ func TestCheckInvalidArrayRemoveLastFromConstantSized(t *testing.T) {
 	assert.IsType(t, &sema.NotDeclaredMemberError{}, errs[0])
 }
 
+func TestCheckArrayIndexOf(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+      fun test(): Int? {
+          let x = [1, 2, 3]
+					return x.firstIndex(of:2)
+      }
+    `)
+
+	require.NoError(t, err)
+}
+
+func TestCheckArrayFirstIndexDoesNotExist(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+      fun test(): Int? {
+          let x = [1, 2, 3]
+					return x.firstIndex(of:5)
+      }
+    `)
+
+	require.NoError(t, err)
+}
+
+func TestCheckArrayFirstIndexWrongType(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+      fun test(): Int? {
+          let x = [1, 2, 3]
+					return x.firstIndex(of:"foo")
+      }
+    `)
+	errs := ExpectCheckerErrors(t, err, 1)
+	assert.IsType(t, &sema.TypeMismatchError{}, errs[0])
+}
+
+func TestCheckInvalidResourceFirstIndex(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+      resource X {}
+
+      fun test(): Int? {
+          let xs <- [<-create X()]
+					return <-xs.firstIndex(of: "foo")
+      }
+    `)
+
+	errs := ExpectCheckerErrors(t, err, 7)
+
+	assert.IsType(t, &sema.InvalidResourceArrayMemberError{}, errs[0])
+	assert.IsType(t, &sema.NotEquatableTypeError{}, errs[1])
+	assert.IsType(t, &sema.TypeMismatchError{}, errs[2])
+	assert.IsType(t, &sema.MissingMoveOperationError{}, errs[3])
+	assert.IsType(t, &sema.InvalidMoveOperationError{}, errs[4])
+	assert.IsType(t, &sema.ResourceLossError{}, errs[5])
+	assert.IsType(t, &sema.ResourceLossError{}, errs[6])
+}
+
```

Comment:
Could you please also add a checker test for a non-equatable type here? For example, arrays are not equatable, so testing an array of arrays would be suitable

Reply:
I added test with array of arrays.